ncm-filesystems/src/main/perl/filesystems.pm

```diff
@@ -1,116 +1,143 @@
 #${PMpre} NCM::Component::filesystems${PMpost}
-
-use LC::Exception qw (throw_error);
-use LC::Process qw (output);
-use Cwd qw(abs_path);
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+package NCM::Component::filesystems;
+
+use strict;
+use warnings;
+use NCM::Component;
+use CAF::Process;
+use CAF::FileEditor;
+use CAF::FileReader;
 use NCM::Filesystem;
 use NCM::Partition qw (partition_compare);
-use CAF::Object;
-
-use constant PROTECTED_PATH => "/software/components/fstab/protected_mounts";
-
-use parent qw(NCM::Component);
-our $EC = LC::Exception::Context->new->will_store_all;
+use Fcntl qw(:seek);
+use Readonly;
+use Cwd qw(abs_path);
 
-# Returns a hash with the canonical paths to the protected mountpoints
-# as its keys. I This small overhead may simplify the code later.
-sub protected_hash
-{
-	my $config = shift;
-	my %ph;
+Readonly my $UMOUNT => "/bin/umount";
+Readonly my $FSTAB_CMP_PATH => '/software/components/fstab';
+Readonly my $FS_TREE => '/system/filesystems';
+use parent qw(NCM::Component::fstab);
 
-	my $pl = $config->getElement (PROTECTED_PATH)->getTree;
-
-	foreach my $i (@$pl) {
-		$ph{$i} = 0;
-		my $path = abs_path("$i/");
-		$ph{$path} = 0 if $path;
-	}
-	return %ph;
-}
+our $EC = LC::Exception::Context->new()->will_store_all();
+our $NoActionSupported = 0; # First needs check/fixing in ncm-lib-blockdevices issue #74
```

Comment:
issue #74 is created for that. it is not really a requirement for this PR (I moved the NoAction test, but filesystems NoAction behaviour isn't changed here).